backend/src/sse/broadcaster.rs

```diff
@@ -0,0 +1,156 @@
+//! SSE broadcaster.
+
+use actix_web_lab::sse::{self, ChannelStream, Sse};
+use futures_util::future;
+use std::{collections::HashMap, sync::Arc, time::Duration};
+use tokio::{sync::Mutex, time::interval};
+
+/// Inner state of SSE broadcaster.
+#[derive(Debug, Clone)]
+struct BroadcasterInner {
+    /// Map of map_id to a connected map.
+    /// The map_id is the id of the map that the client connected to.
+    /// The connected map contains the map_id and a list of clients connected to that map.
+    maps: HashMap<i32, ConnectedMap>,
+}
+
+/// Map that clients are connected to.
+#[derive(Debug, Clone)]
+struct ConnectedMap {
+    map_id: i32,
+    clients: Vec<ConnectedClient>,
+}
+
+/// Client listening to updates on a map.
+#[derive(Debug, Clone)]
+pub struct ConnectedClient {
+    /// Client ID.
+    id: String,
+    /// SSE sender for writing messages to the client.
+    sender: sse::Sender,
+}
+
+/// SSE broadcaster.
+pub struct Broadcaster {
+    /// Inner state of SSE broadcaster, guarded by a mutex.
+    inner: Mutex<BroadcasterInner>,
+}
+
+impl Broadcaster {
+    /// Constructs new broadcaster and spawns ping loop.
+    #[must_use]
+    pub fn create() -> Arc<Self> {
+        let this = Arc::new(Self {
+            inner: Mutex::new(BroadcasterInner {
+                maps: HashMap::new(), // TODO: how can we choose a better initial capacity?
+            }),
+        });
+
+        Self::spawn_ping(Arc::clone(&this));
+        this
+    }
+
+    /// Pings clients every 10 seconds to see if they are alive and remove them from the broadcast list if not.
```

Comment:
Imho this is much too frequent. Why do we need this at all? Can we implement this together with auth renewal to not have separate pings?